lightningd/params.h

```diff
@@ -89,4 +90,16 @@ typedef bool(*param_cb)(const char *buffer, const jsmntok_t *tok, void *arg);
 		      (arg) + 0*sizeof(*(arg) == (jsmntok_t *)NULL),	\
 		      sizeof(const jsmntok_t *)
 
+/*
+ * Convenient way to set an optional @param to a @value if it wasn't set during
+ * parsing.
+ */
+#define param_set_if_null(ctx, param, value)              \
```

Comment:
Hmm, can we do something like param_opt_default() and get this into one line?

Seems to work?

     https://github.com/rustyrussell/lightning/tree/param-with-defaults

Reply:
Yes it works great.  I have it running locally and here's what it looks like:
```C
if (!param_parse(cmd, buffer, params,
                 param_req("msatoshi", json_tok_tok, &msatoshi),
                 param_req("label", json_tok_tok, &label),
                 param_req("description", json_tok_tok, &desctok),
                 param_opt_default("expiry", json_tok_u64, &expiry, 3600),
                 param_opt_tok("fallback", &fallback),
                 param_opt_tok("fallbacks", &fallbacks),
                 param_opt_tok("preimage", &preimagetok), NULL))
        return;
```
What do you think of shortening the macro names? Something like:
```C
if (!param_parse(cmd, buffer, params,
                 p_req("msatoshi", json_tok_tok, &msatoshi),
                 p_req("label", json_tok_tok, &label),
                 p_req("description", json_tok_tok, &desctok),
                 p_opt_def("expiry", json_tok_u64, &expiry, 3600),
                 p_opt_tok("fallback", &fallback),
                 p_opt_tok("fallbacks", &fallbacks),
                 p_opt_tok("preimage", &preimagetok), NULL))
        return;
```

Reply:
Agreed, good idea. Rename param_parse to params() too?

Reply:
Yes.

Reply:
I can take care of all this if you like.